### ISSUE TITLE
Unable to pull audio & screenshot from funimation

### ISSUE THREAD

**Issue by Jarods-language-journey** [Ajatt-Tools/mpvacious#29]
**Environment:**
OS: Windows 10
Update: Latest windows 10
Latest mpv version - 0.33.0 (2-28-21)

**Describe the bug**
Using a chrome browser extension "watch with MPV," Mpvacious doesn't seem to be able to extract audio while being played through MPV.  The site from which I am watching from is Funimation.  I can watch it completely fine, it just doesn't grab onto the audio & image being played.

**To Reproduce**
1. Open video using "watch with MPV" extension
2. Select word from clipboard monitor
3. Create and update Anki card using "c" and "m"
4. Open anki card
5. Preview anki card

**Expected behavior**
A playthrough of the audio and a screenshot of the image to be in the preview

**Screenshots**

![image](https://user-images.githubusercontent.com/80096108/111728190-9cc31780-8829-11eb-9111-ed959790096e.png)
![image](https://user-images.githubusercontent.com/80096108/111728227-b2384180-8829-11eb-91be-a8f0e55ab68f.png)

If I try to watch directly through cmd, this is what I get so I use the chrome extension instead.

`C:\Users\jarod>mpv https://www.funimation.com/shows/log-horizon/the-apocalypse/?qid=undefined
C:\Users\jarod\AppData\Roaming/mpv/mpv.conf:3: unparseable extra characters: ':jpn,jp,en'
C:\Users\jarod\AppData\Roaming/mpv/mpv.conf:4: unparseable extra characters: ':jpn,jp,en'
Resuming playback. This behavior can be disabled with --no-resume-playback.
[ytdl_hook] ERROR: Unable to extract al:web:url; please report this issue on https://yt-dl.org/bug . Make sure you are using the latest version; type  youtube-dl -U  to update. Be sure to call youtube-dl with the --verbose flag and include its complete output.
[ytdl_hook] youtube-dl failed: unexpected error occurred
Failed to recognize file format.

Exiting... (Errors when loading file)
`

**Comment by tatsumoto-ren** [comment]
mpvacious retrieves the video's filename from mpv. In your case it says

> quiundefined

That's very mysterious.

**Comment by Jarods-language-journey** [comment]
If I may add on, the end of the funimation url is "qid=undefined".

I believe this may have something to do with how funimation and having anti robot detection but if it's something else I'm not sure.

**Comment by tatsumoto-ren** [comment]
Card creation will work if the stream url is supplied to the mpv encoder. You need to figure out how to retrieve it. The appropriate changes then could be applied to the add-on's code.

**Comment by Jarods-language-journey** [comment]
Oh I see what you mean. Lastly, by add-on, do you mean the chrome extension or the mpvacious add-on?  

From what I'm getting, I'll have to find out is how to supply that stream url to mpv so mpvacious can extract it then from mpv.

**Comment by tatsumoto-ren** [comment]
I mean this add-on. If we figure out how mpv gets the stream url it might be possible to automate card creation.

**Comment by SecondaryH** [comment]
I've actually been encountering the same issue but with Plex playing through mpv.

Screenshots:
![image](https://user-images.githubusercontent.com/76407423/115971412-9cb0e880-a548-11eb-8ada-b559ae2036e0.png)

![image](https://user-images.githubusercontent.com/76407423/115971416-a5092380-a548-11eb-9300-d6d56b8c64f6.png)

![image](https://user-images.githubusercontent.com/76407423/115971423-a9cdd780-a548-11eb-960b-242f9cbe2255.png)
(I've blurred out my plex auth token)

Snapshots are not made and the audio doesn't play when clicking the play button.

**Comment by tatsumoto-ren** [comment]
I assume the media files fail to get created. The reason lies somewhere in the fact that the source is a remote url.

**Comment by SecondaryH** [comment]
Yeah I guess its something that I'll have to accept as a con. Atleast I can manually add an image for context

**Comment by Jarods-language-journey** [comment]
**Environment:**
OS: Windows 10
Update: Latest windows 10
Latest mpv version - 0.33.0 (2-28-21)

**Bug**
No audio and no snapshot.

I'm coming back to this topic as I've encountered some errors with password protected streams (on VRV).  Is there a way to obtain the audio and image of a password protected stream to MPV using mpvacious?  When using mpvacious, it seems that it doesn't use my mpv.conf settings as I have ytdl settings that provide VRV with my username, password and formatting settings.

MPV Log:
```
[ 115.203][d][cplayer] Run command: subprocess, flags=64, args=[args="curl,-s,localhost:8765,-H,Content-Type: application/json; charset=UTF-8,-X,POST,--data-binary,@C:\\Users\\jarod\\AppData\\Local\\Temp/curl_tmp.txt", playback_only="no", capture_size="67108864", capture_stdout="yes", capture_stderr="no", detach="no", env="", stdin_data="", passthrough_stdin="no"]
[ 115.467][e][mpvacious_master] Couldn't store file 'TwoNationsoftheStoneWorld_01m57s242ms.webp': [Errno 2] No such file or directory: 'C:\\Users\\jarod\\AppData\\Local\\Temp/TwoNationsoftheStoneWorld_01m57s242ms.webp' 
[ 115.467][d][cplayer] Run command: subprocess, flags=64, args=[args="curl,-s,localhost:8765,-H,Content-Type: application/json; charset=UTF-8,-X,POST,--data-binary,@C:\\Users\\jarod\\AppData\\Local\\Temp/curl_tmp.txt", playback_only="no", capture_size="67108864", capture_stdout="yes", capture_stderr="no", detach="no", env="", stdin_data="", passthrough_stdin="no"]
[ 115.742][e][mpvacious_master] Couldn't store file 'TwoNationsoftheStoneWorld_01m56s370ms-01m58s960ms.ogg': [Errno 2] No such file or directory: 'C:\\Users\\jarod\\AppData\\Local\\Temp/TwoNationsoftheStoneWorld_01m56s370ms-01m58s960ms.ogg' 
```

**Comment by tatsumoto-ren** [comment]
https://tatsumoto.neocities.org/blog/mining-from-movies-and-tv-shows.html#download-content
